backend/compact-connect/pipeline/backend_pipeline.py

```diff
@@ -98,3 +105,22 @@ def build_pipeline(self) -> None:
             ],
             apply_to_children=True
         )
+
+        self._add_alarms()
+
+    def _add_alarms(self):
+        NotificationRule(
+            self, 'NotificationRule',
+            source=self.pipeline,
+            events=[
+                'codepipeline-pipeline-pipeline-execution-started',
+                'codepipeline-pipeline-pipeline-execution-failed',
+                'codepipeline-pipeline-pipeline-execution-succeeded',
+                'codepipeline-pipeline-manual-approval-needed'
```

Comment:
What's the naming scheme here? (Trying to parse why `pipeline` is written out three times :D

Reply:
These are just the [names that CodePIpeline defines](https://docs.aws.amazon.com/dtconsole/latest/userguide/concepts.html#events-ref-pipeline)